src/data_provider/tests/sysInfoPackageLinuxParserRpm/sysInfoPackageLinuxParserRPM_test.cpp

```diff
@@ -0,0 +1,504 @@
+            int32_t* tmp = (int32_t*)cp;
+            *tmp = __builtin_bswap32(tag);
+            cp += sizeof(int32_t);
+            // type
+            tmp = (int32_t*)cp;
+            *tmp = __builtin_bswap32(type);
+            cp += sizeof(int32_t);
+            //offset
+            tmp = (int32_t*)cp;
+            *tmp = __builtin_bswap32(bytes_count);
+            bytes_count += len;
+            cp += sizeof(int32_t);
+            // unused data
+            cp += sizeof(int32_t);
+        }
+    };
+
+
+    auto content_string
+    {
+        [&cp](std::string value)
+        {
+            strcpy(cp, value.c_str());
+            cp += value.length() + 1;
+        }
+    };
+    auto content_int
+    {
+        [&cp](int value)
+        {
+            int32_t* tmp = (int32_t*)cp;
+            *tmp = __builtin_bswap32(value);
+            cp += sizeof(int);
+        }
+    };
+
+
+    {
+        // Header
+        // index lenght
+        ip = (int32_t*)cp;
+        *ip = __builtin_bswap32(total_fields); // TAG_X COUNT
+        cp += sizeof(int);
+
+        // Data lenght
+        ip = (int32_t*)cp;
+        *ip = __builtin_bswap32(total_fields_len);
+        cp += sizeof(int);
+    }
+
+    entry(TAG_NAME, STRING_TYPE, name.length() + 1);
+    entry(TAG_VERSION, STRING_TYPE, version.length() + 1);
+    entry(TAG_RELEASE, STRING_TYPE, release.length() + 1);
+    entry(TAG_EPOCH, INT32_TYPE, sizeof(epoch));
+    entry(TAG_SUMMARY, STRING_TYPE, summary.length() + 1);
+    entry(TAG_ITIME, INT32_TYPE, sizeof(itime));
+    entry(TAG_SIZE, INT32_TYPE, sizeof(size));
+    entry(TAG_VENDOR, STRING_TYPE, vendor.length() + 1);
+    entry(TAG_GROUP, STRING_TYPE, group.length() + 1);
+    entry(TAG_SOURCE, STRING_TYPE, source.length() + 1);
+    entry(TAG_ARCH, STRING_TYPE, arch.length() + 1);
+
+    content_string(name);
+    content_string(version);
+    content_string(release);
+    content_int(epoch);
+    content_string(summary);
+    content_int(itime);
+    content_int(size);
+    content_string(vendor);
+    content_string(group);
+    content_string(source);
+    content_string(arch);
+
+
+    EXPECT_CALL(*libdb_mock, c_get(_, _, _, _)).Times(3)
+    .WillOnce(DoAll(SetArgPointee<1>(key), SetArgPointee<2>(data), Return(0)))
+    .WillOnce(DoAll(SetArgPointee<1>(key), SetArgPointee<2>(data), Return(0)))
+    .WillOnce(Return(1));
+    EXPECT_CALL(*libdb_mock, c_close(_)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(*libdb_mock, close(_, _)).Times(1).WillOnce(Return(0));
+
+
+    EXPECT_CALL(wrapper, callbackMock(expectedPackage1)).Times(1);
+
+    getRpmInfo([&wrapper](nlohmann::json & packageInfo)
+    {
+        wrapper.callbackMock(packageInfo);
+    });
+
+}
+
+
+TEST(SysInfoPackageLinuxParserRPM_test, rpmFromLibRPM)
+{
+    CallbackMock wrapper;
+
+    auto expectedPackage1 = R"({"name":"1","architecture":"2","description":"3","size":4,"version":"5:7-6","vendor":"8","install_time":"9","groups":"10","format":"rpm"})"_json;
```

Comment:
Hi @jnasselle 

I think the **expectedPackage1** variable should be const

Best regards
Hanes